omnigan/trainer.py

```diff
@@ -1614,3 +1618,43 @@ def del_output_path(self, force=False):
 
         if (Path(self.opts.output_path) / "is_functional.test").exists() or force:
             shutil.rmtree(self.opts.output_path)
+
+    def compute_smog(self, x, z=None, d=None, s=None, use_sky_seg=False):
+        sky_mask = None
+        if d is None or (use_sky_seg and s is None):
+            if z is None:
+                z = self.G.encode(x)
+            if d is None:
+                d = self.G.decoders["d"](z)
+            if use_sky_seg and s is None:
+                s = self.G.decoders["s"](z)
+                # todo: s to sky mask
+                # todo: interpolate to d's size
+
+        airlight = 0.76
+        A = airlight * torch.ones(3).view(1, -1, 1, 1)
+        I = srgb2lrgb(x)
+        vr = 1
+        beta_param = 2
+
+        beta = torch.tensor([beta_param / vr, beta_param / vr, beta_param / vr]).view(
+            1, -1, 1, 1
+        )
+        d = normalize(d, mini=0.3, maxi=1.0)
+        d = 1.0 / d
```

Comment:
What is this inversion for?

Reply:
Because the values stored are actually inverse depth, so that the sky needs to be predicted at 0 and not +inf and there is no inf problem because no object has a depth of 0 (which could theoretically yield +inf)